pallets/loan/src/lib.rs

```diff
@@ -819,6 +977,96 @@ impl<T: Config> Pallet<T> {
 		);
 		Ok(nav)
 	}
+
+	/// writes off a given unhealthy loan
+	/// if override_write_off_index is Some, this is a admin action and loan override flag is set
+	/// if loan is already overridden and override_write_off_index is None, we return error
+	/// if loan is still healthy, we return an error
+	/// loan is accrued and nav is updated accordingly
+	/// returns new write off index applied to loan
+	fn write_off(
+		pool_id: T::PoolId,
+		loan_id: T::LoanId,
+		override_write_off_index: Option<u32>,
+	) -> Result<u32, DispatchError> {
+		LoanInfo::<T>::try_mutate(
+			pool_id,
+			loan_id,
+			|maybe_loan_data| -> Result<u32, DispatchError> {
+				let mut loan_data = maybe_loan_data.take().ok_or(Error::<T>::ErrMissingLoan)?;
+				// ensure loan is active
+				ensure!(
+					loan_data.status == LoanStatus::Active,
+					Error::<T>::ErrLoanNotActive
+				);
+
+				let maturity_date = loan_data
+					.loan_type
+					.maturity_date()
+					.ok_or(Error::<T>::ErrLoanTypeInvalid)?;
+				let now = Self::time_now()?;
+
+				// ensure loan's maturity date has passed
```

Comment:
Ideally, write-offs should even be possible pre maturity date. in Tinlake on Ethereum this wasn't allowed, to simplify, but it would be really nice if we don't have to have this restriction in TOCC. I think the way the spec was written, this should not be an issue, this check can simply be removed.

Reply:
do you mean when admin write off the loan?

Reply:
Yeah indeed, it should ideally be possible for an admin to write off a loan before the maturity date.